MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/extrusions/RotationExtrusionActivity.java

```diff
@@ -0,0 +1,182 @@
+  //90 is used based on the viewable angle when viewing the map (from phone being flat to facing you
+  private int pitchAmplifier = -90;
+  private int bearingAmplifier = 90;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_extrusion_rotation);
+
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+        @Override
+        public void onMapReady(@NonNull final MapboxMap map) {
+          mapboxMap = map;
+          setupBuildings();
+        }
+    });
+
+    //initialize sensors
+    sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+
+    gyro = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+    magnetic = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD);
+
+
+  }
+
+  private void setupBuildings() {
+    FillExtrusionLayer fillExtrusionLayer = new FillExtrusionLayer("3d-buildings", "composite");
+    fillExtrusionLayer.setSourceLayer("building");
+    fillExtrusionLayer.setFilter(eq("extrude", "true"));
+    fillExtrusionLayer.setMinZoom(15);
+    fillExtrusionLayer.setProperties(
+            fillExtrusionColor(Color.LTGRAY),
+            fillExtrusionHeight(Function.property("height", new IdentityStops<Float>())),
+            fillExtrusionBase(Function.property("min_height", new IdentityStops<Float>())),
+            fillExtrusionOpacity(0.9f)
+    );
+    mapboxMap.addLayer(fillExtrusionLayer);
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+    sensorManager.registerListener(this, gyro , 200);
+    sensorManager.registerListener(this, magnetic , 200);
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+    sensorManager.unregisterListener(this, gyro);
+    sensorManager.unregisterListener(this, magnetic);
+  }
+
+  @Override
+  public void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  public void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  @Override
+  public void onSensorChanged(SensorEvent event) {
```

Comment:
`onSensorChanged` is a bit long.
What about simplifying it extracting blocks of code into `private` methods and make intention clearer?